maven-plugin/src/main/resources/META-INF/plexus/components.xml

```diff
@@ -0,0 +1,124 @@
+<!--
+  ~ Copyright The WildFly Authors
+  ~ SPDX-License-Identifier: Apache-2.0
+  -->
+
+<!--
+   Define Plexus components to allow this plugin to function as
+   a Maven extension that defines new packaging types 'galleon-feature-pack' and 'galleon-user-feature-pack'.
+   When building a module with these types the
+-->
+<component-set>
+  <components>
+    <component>
+      <role>org.apache.maven.lifecycle.mapping.LifecycleMapping</role>
+      <role-hint>galleon-feature-pack</role-hint>
+      <implementation>org.apache.maven.lifecycle.mapping.DefaultLifecycleMapping</implementation>
+      <configuration>
+        <lifecycles>
+          <lifecycle>
+            <id>default</id>
+            <phases>
+              <process-resources>
+                org.apache.maven.plugins:maven-resources-plugin:resources
+              </process-resources>
+              <compile>
+                org.apache.maven.plugins:maven-compiler-plugin:compile
+              </compile>
+              <process-test-resources>
+                org.apache.maven.plugins:maven-resources-plugin:testResources
+              </process-test-resources>
+              <test-compile>
+                org.apache.maven.plugins:maven-compiler-plugin:testCompile
+              </test-compile>
+              <test>
+                org.apache.maven.plugins:maven-surefire-plugin:test
+              </test>
+              <package>
+                <!-- Replace maven-jar-plugin -->
+                org.wildfly.galleon-plugins:wildfly-galleon-maven-plugin:build-feature-pack
+              </package>
+              <install>
+                org.apache.maven.plugins:maven-install-plugin:install
+              </install>
+              <deploy>
+                org.apache.maven.plugins:maven-deploy-plugin:deploy
+              </deploy>
+            </phases>
+          </lifecycle>
+        </lifecycles>
+      </configuration>
+    </component>
+    <component>
```

Comment:
Is this component really needed? I experimented building FP without it and all went fine.

Reply:
@jfdenise I'll try again. In earlier iterations of trying this kind of thing (not this plugin; a POC one) it wasn't working, but then i found https://github.com/spotify/docker-maven-plugin/blob/master/src/main/resources/META-INF/plexus/components.xml and added this bit based on it. But other docs I've seen of this kind of thing didn't have this so maybe I was doing something else wrong.

Reply:
I removed it. This discussion is now pointing to a different logical line than it was before. :) The component it now points to should remain.

Reply:
Please hold on merging this!

I removed this and now in the maven repo the artifact has the wrong name. For example:

wildfly-mvc-krazo-feature-pack-2.0.0.Alpha4.galleon-feature-pack.

So I think the deleted components were important!

Reply:
Ok, I've restored these. The maven repo contents are as expected now.